scripts/cybershake/cybershake_progress.py

```diff
@@ -0,0 +1,364 @@
+
+    fault_names, r_counts = [], []
+    for ix, line in enumerate(lines):
+        line_l = line.split(" ")
+        fault_names.append(line_l[0].strip())
+
+        try:
+            r_counts.append(int(line_l[1].strip().rstrip("RrNnKk")))
+        except ValueError:
+            print(
+                "Failed to read line {} of cybershake list. Need to know"
+                "the number of realisations for each fault, quitting!".format(ix + 1)
+            )
+            # Raise the exception so execution stops
+            raise
+
+    return np.asarray(fault_names), np.asarray(r_counts)
+
+
+def get_new_progress_df(root_dir, runs_dir, faults, fault_names, r_counts):
+    """Gets a new progress dataframe, runs the full estimation + collects
+    all actual core hours.
+    """
+    # Run the estimation
+    est_args = argparse.Namespace(
+        vms_dir=sim_struct.get_VM_dir(root_dir),
+        sources_dir=sim_struct.get_sources_dir(root_dir),
+        runs_dir=runs_dir,
+        fault_selection=None,
+        root_yaml=None,
+        output=None,
+        verbose=False,
+        models_dir=None,
+    )
+    df = est_cybershake(est_args)
+    grouped_df = df.groupby("fault_name").sum()
+    grouped_df.sort_index(axis=0, level=0, inplace=True)
+
+    # Sanity check
+    if np.any(grouped_df.index.values != fault_names):
+        raise Exception(
+            "The fault names of the estimation results and directories "
+            "are not matching. These have to match, quitting!"
+        )
+
+    # Create progress dataframe
+    column_t = []
+    for proc_type in PROCESS_TYPES:
+        column_t.append((proc_type, EST_CORE_HOURS_COL))
+        column_t.append((proc_type, ACT_CORE_HOURS_COL))
+    column_t.append(("total", EST_CORE_HOURS_COL))
+    column_t.append(("total", ACT_CORE_HOURS_COL))
+
+    progress_df = pd.DataFrame(
+        index=grouped_df.index.values, columns=pd.MultiIndex.from_tuples(column_t)
+    )
+
+    # Populate progress dataframe with estimation data
+    total = np.nan
+    for proc_type in PROCESS_TYPES:
+        values = grouped_df[proc_type, const.MetadataField.core_hours.value]
+        progress_df[proc_type, EST_CORE_HOURS_COL] = values
+        total = values if total is np.nan else total + values
+    progress_df["total", EST_CORE_HOURS_COL] = total
+    progress_df["total", R_COUNT_COL] = r_counts
+    # Get actual core hours for all faults, assumes faults are run
+    # in alphabetical order
+    chours_df = get_chours_used(faults)
+
+    # Add actual data to progress df
+    total = np.nan
+    for proc_type in PROCESS_TYPES:
+        values = chours_df[proc_type]
+        progress_df[proc_type, ACT_CORE_HOURS_COL] = values
+        total = values if total is np.nan else total + values
+    progress_df["total", ACT_CORE_HOURS_COL] = total
+
+    return progress_df
+
+
+def load_progress_df(file: str):
+    return pd.read_csv(file, index_col=[0], header=[0, 1])
+
+
+def print_progress(
```

Comment:
Does this function print or write to a csv file?

Reply:
Print to a outfile, and also dumps dataframe into a CSV. I think dumping to CSV is unnecessary, so will remove it.

Reply:
Print to an outfile? Do you mean printing to stdout or writing to outfile?

Reply:
writing to outfile - will update comments suggesting it is "printing"